contracts/src/tokens/drop/DropFacetBase.sol

```diff
@@ -0,0 +1,178 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+// interfaces
+import {IDropFacetBase} from "./IDropFacet.sol";
+
+// libraries
+import {DropStorage} from "./DropStorage.sol";
+import {CustomRevert} from "contracts/src/utils/libraries/CustomRevert.sol";
+import {MerkleProofLib} from "solady/utils/MerkleProofLib.sol";
+
+// contracts
+
+abstract contract DropFacetBase is IDropFacetBase {
+  using DropStorage for DropStorage.Layout;
+  using MerkleProofLib for bytes32[];
+
+  function _getActiveConditionId(
+    DropStorage.Layout storage ds
+  ) internal view returns (uint256) {
+    uint256 conditionStartId = ds.conditionStartId;
+    uint256 conditionCount = ds.conditionCount;
+
+    if (conditionCount == 0) {
+      CustomRevert.revertWith(DropFacet__NoActiveClaimCondition.selector);
+    }
+
+    uint256 currentTimestamp = block.timestamp;
```

Comment:
Caching `block.timestamp` to a stack variable is in fact not desirable because the [timestamp opcode](https://www.evm.codes/?fork=cancun#42) only cost 2 gas while stack opcodes cost 3.

Reply:
nice! thanks!